#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/traffic/protos/TrafficCone.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/traffic/protos/DirectionPanel.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.39432965953419635 0.370499620221898 -0.8409721464038356 1.7383124681632194
  position -0.2296486008844817 5.012760317472732 5.100250312941313
}
TexturedBackgroundLight {
  castShadows FALSE
}
RectangleArena {
  rotation 1 0 0 0
  floorSize 6 6
  floorAppearance PBRAppearance {
    baseColor 0 0.1 0.1
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
}
DEF Automotive E-puck {
  translation -0.19947 2.47346 -6.3962e-05
  rotation 5.9525e-10 1.60483e-09 1 -1.5771953071795863
  name "Automotive"
  controller "AutomotiveController"
  groundSensorsSlot [
    DistanceSensor {
      translation 0 0 0.05
      name "cone_sensor"
    }
    DistanceSensor {
      translation 0 0 0.005
      name "obstacle_sensor"
      type "infra-red"
    }
    Emitter {
      name "ChildParentEmitter"
    }
    Receiver {
      name "ParentChildReceiver"
      channel 1
    }
    Robot {
      translation 0.01 0 0.04
      children [
        Emitter {
          name "ParentChildEmitter"
          channel 1
        }
        Receiver {
          name "ChildParentReceiver"
        }
        Camera {
          translation 0.01 0 0
          name "CAM"
          width 1280
          height 720
        }
      ]
      name "Raspberry"
      controller "RaspberryController"
      controllerArgs [
        "C"
      ]
    }
    DistanceSensor {
      translation 0.04 -0.03 0.01
      rotation 0 1 0 1.5708
      name "IR0"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 -0.02 0.01
      rotation 0 1 0 1.5708
      name "IR1"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 -0.01 0.01
      rotation 0 1 0 1.5708
      name "IR2"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 0 0.01
      rotation 0 1 0 1.5708
      name "IR3"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 0.01 0.01
      rotation 0 1 0 1.5708
      name "IR4"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 0.02 0.01
      rotation 0 1 0 1.5708
      name "IR5"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.04 0.03 0.00999996
      rotation 0 1 0 1.5708
      name "IR6"
      type "infra-red"
    }
  ]
}
TrafficCone {
  translation -0.19 0.04 0
  scale 0.2
  name "traffic cone(1)"
}
Solid {
  translation -2.21 2.26 -0.016
  children [
    Transform {
      translation -0.19 -0.01 0.01
      scale 0.0007 0.0007 0.0007
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../assets/01-02-Bodenmuster/Bodenmuster-01-02-B.obj"
            ]
          }
        }
      ]
    }
  ]
  name "Wegnetz"
}
DirectionPanel {
  translation 1.91874 -1.7256 -0.02
  rotation -0.2505618966828277 -0.9351136144134614 0.2505618966828277 -1.6378353071795866
  frontTextTexture [
    "../assets/01-02-Bodenmuster/as.46321139.jpg"
  ]
}
DirectionPanel {
  translation -0.595322 -2.61281 -0.02
  rotation 0.577349935856137 0.5773509358560258 -0.577349935856137 2.09439
  name "direction panel(1)"
  frontTextTexture [
    "../assets/01-02-Bodenmuster/images.png"
  ]
}
DirectionPanel {
  translation -2.63964 -1.08765 -0.02
  rotation 0.6785984389637569 0.2810841818244213 -0.6785984389637569 2.59357
  name "direction panel(2)"
  frontTextTexture [
    "../assets/01-02-Bodenmuster/buchstabe-c-fuer-hausnummernleuchte-schwarz-selbstklebend.jpg"
  ]
}
Solid {
  translation 0.31721 1.2055 3.10863e-15
  rotation 0 0 1 -1.5707953071795862
  children [
    Transform {
      scale 0.0005 0.0005 0.0005
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Mesh {
            url [
              "../assets/01-01-Hindernis/Hindernis-01-01-A.obj"
            ]
          }
        }
      ]
    }
  ]
  name "Hindernis Rot"
}
Solid {
  translation 0.701125 0.54086 3.10863e-15
  rotation 0 0 1 2.0944
  children [
    Transform {
      scale 0.0005 0.0005 0.0005
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Mesh {
            url [
              "../assets/01-01-Hindernis/Hindernis-01-01-A.obj"
            ]
          }
        }
      ]
    }
  ]
  name "Hindernis Rot(1)"
}
DEF hide-line Solid {
  translation 0.778693 0.611243 -0.01
  rotation 0 0 1 -1.047195307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.16 0.005
      }
    }
  ]
}
DEF hide-line Solid {
  translation 0.453693 1.17702 -0.02
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.51 0.005
      }
    }
  ]
  name "solid(1)"
}
DEF hide-line Solid {
  translation 0.778693 1.73992 -1.73472e-18
  rotation 0 0 1 -2.094395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(2)"
}
DEF hide-line Solid {
  translation 1.75869 0.04992 -0.02
  rotation 0 0 1 3.14159
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(3)"
}
DEF hide-line Solid {
  translation 0.776485 -1.64877 -0.01
  rotation 0 0 1 -1.047195307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(4)"
}
DEF hide-line Solid {
  translation -1.18075 -1.63879 -0.01
  rotation 0 0 1 -2.094395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(5)"
}
DEF hide-line Solid {
  translation -2.16075 0.04121 0
  rotation 0 0 1 3.14159
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(6)"
}
DEF hide-line Solid {
  translation -1.16989 1.74491 -0.03
  rotation 0 0 -1 -2.094395307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.165 0.005
      }
    }
  ]
  name "solid(7)"
}
DEF hide-line Solid {
  translation -0.524925 1.74203 0
  rotation 0 0 1 2.61799
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 1.21 0.005
      }
    }
  ]
  name "solid(8)"
}
DEF hide-line Solid {
  translation -1.50492 1.18203 -0.02
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 1.21 0.005
      }
    }
  ]
  name "solid(9)"
}
DEF hide-line Solid {
  translation -0.52996 0.613346 -0.01
  rotation 0 0 1 0.5236
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 1.21 0.005
      }
    }
  ]
  name "solid(10)"
}
DEF hide-line Solid {
  translation -1.50494 0.0446339 0
  rotation 0 0 1 -0.523595307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.52 0.005
      }
    }
  ]
  name "solid(11)"
}
DEF hide-line Solid {
  translation -1.17994 -0.51827 -0.01
  rotation 0 0 1 -1.047195307179586
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.17 0.005
      }
    }
  ]
  name "solid(12)"
}
DEF hide-line Solid {
  translation -0.19994 -1.07827 -0.01
  rotation 0 0 1 2.24093e-06
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.17 0.005
      }
    }
  ]
  name "solid(13)"
}
DEF hide-line Solid {
  translation 0.783683 -0.514576 -0.01
  rotation 0 0 1 1.0472
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.1
        }
      }
      geometry Box {
        size 0.03 2.17 0.005
      }
    }
  ]
  name "solid(14)"
}
